src/fal/planner/plan.py

```diff
@@ -41,16 +66,70 @@ def from_execution_plan(
 
 
 @dataclass
-class PlannedGraph(OriginGraph):
+class ScriptConnectedGraph(OriginGraph):
     graph: nx.DiGraph
 
     @classmethod
     def from_filtered_graph(
         cls,
         filtered_graph: FilteredGraph,
-        enable_chunking: bool = True,
     ):
         graph = filtered_graph.copy_graph()
+        shuffled_graph = cls(graph)
+        shuffled_graph._shuffle()
+        return shuffled_graph
+
+    def _shuffle(self):
+        def _pattern_matching(node_check: Callable[[str], bool], nodes: Set[str]):
+            matched = {
+                maybe_script for maybe_script in nodes if node_check(maybe_script)
+            }
+            return matched, nodes - matched
+
+        def get_before_scripts(graph: nx.DiGraph, node: str):
+            return _pattern_matching(_is_before_script, set(graph.predecessors(node)))
+
+        def get_after_scripts(graph: nx.DiGraph, node: str):
+            return _pattern_matching(_is_after_script, set(graph.successors(node)))
+
+        def _add_edges_from_to(from_nodes: Set[str], to_nodes: Set[str]):
+            self.graph.add_edges_from(
+                (from_n, to_n) for from_n in from_nodes for to_n in to_nodes
+            )
+
+        old_graph = self.copy_graph()
+        node: str
```

Comment:
We probably don't need this.
```suggestion
```

Reply:
If I comment it:
<img width="272" alt="image" src="https://user-images.githubusercontent.com/2745502/179792616-6279d291-318f-4945-8c31-f1f1a62339ee.png">